openmc/deplete/microxs.py

```diff
@@ -20,168 +22,173 @@
 _valid_rxns.append('fission')
 
 
-class MicroXS(DataFrame):
-    """Microscopic cross section data for use in transport-independent depletion.
-
-    .. versionadded:: 0.13.1
+def get_microxs_and_flux(
+        model: openmc.Model,
+        domains,
+        nuclides: Optional[Iterable[str]] = None,
+        reactions: Optional[Iterable[str]] = None,
+        energies: Optional[Union[Iterable[float], str]] = None,
+        chain_file: Optional[PathLike] = None,
+        run_kwargs=None
+    ) -> Tuple[List[np.ndarray], List[MicroXS]]:
```

Comment:
It is not part of PEP8 but it is something I am advocating for moving forward. I've kind of debated internally in my mind for a while whether it is really worth the trouble. What eventually convinced me that we ought to be adding more type hints is the ability of IDEs (e.g., VS Code) to give _much_ better information on types when those hints are present. This is a huge benefit for developers reading through code, and as you probably know most people spend more time reading code than writing code. How many times have you looked through the depletion module and wondered "what is the type of this variable"? Because I know for me personally that happens all the time. Of course, type hints can be enormously helpful for users too, particularly if they are writing a Python script with OpenMC using a good IDE.